src/dodal/common/data.py

```diff
@@ -0,0 +1,145 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Optional
+
+from aiohttp import ClientSession
+from ophyd_async.core import DirectoryInfo, DirectoryProvider
+from pydantic import BaseModel
+
+
+class DataCollectionIdentifier(BaseModel):
+    # VisitService API
+    """
+    Equivalent to a `Scan Number` or `scan_id`, non-globally unique scan identifier.
+    Should be always incrementing, unique per-visit, co-ordinated with any other
+    scan engines.
+    """
+
+    collectionNumber: int
+
+
+class VisitServiceClientBase(ABC):
+    # VisitService API
+    """
+    Object responsible for I/O in determining collection number
+    """
+
+    @abstractmethod
+    async def create_new_collection(self) -> DataCollectionIdentifier:
+        """Create new collection"""
+
+    @abstractmethod
+    async def get_current_collection(self) -> DataCollectionIdentifier:
+        """Get current collection"""
+
+
+class VisitServiceClient(VisitServiceClientBase):
+    # VisitService API
+    """Client for the VisitService REST API exposed by the GDA Server to co-ordinate unique filenames.
+    While VisitService is embedded in GDA, url is likely to be `ixx-control:8088/api`
+    """
+
+    _url: str
+
+    def __init__(self, url: str) -> None:
+        self._url = url
+
+    async def create_new_collection(self) -> DataCollectionIdentifier:
+        async with ClientSession() as session:
+            async with session.post(f"{self._url}/numtracker") as response:
+                if response.status == 200:
+                    json = await response.json()
+                    return DataCollectionIdentifier.parse_obj(json)
+                else:
+                    raise Exception(response.status)
```

Comment:
Can this be a more specific exception class.

Also please can we make the error message actually descriptive about what it was trying to do.